climada/util/yearsets.py

```diff
@@ -22,223 +22,227 @@
+     occur more than once per sampled year).
 
     Parameters:
-        tot_n_events : int
-            Number of events to be sampled
-        n_input_events : int
-         Number of events contained in given event_impacts object
+        events_per_year : array
+            Number of events per sampled year
+        freqs_orig : array
+            Frequency of each input event
 
     Returns:
-        selected_events : array
-            Uniformaly sampled events (length: tot_n_events)
+        sampling_vect : 2D array
+            The sampling vector specifies how to sample the yimp, it consists of one
+            sub-array per sampled_year, which contains the event_ids of the events used to
+            calculate the annual impacts.
       """
 
-    repetitions = np.ceil(tot_n_events/n_input_events).astype('int')
-    indices = np.tile(np.arange(n_input_events), repetitions)
+    sampling_vect = []
 
-    rng = default_rng()
-    selected_events = rng.choice(indices, size=tot_n_events,
-                                 replace=False).astype('int')
+    indices_orig = np.arange(len(freqs_orig))
 
+    freqs = freqs_orig
+    indices = indices_orig
 
-    return selected_events
+    #if check_doubling true: check that every event doesn't occur more than once per sampled year
+    check_doubling = False
 
-def compute_annual_impacts(event_impacts, sampling_dict):
-    """Sample annual impacts from the given event_impacts according to the sampling dictionary
+    #sample events for each sampled year
+    for idx_year, amount_events in enumerate(events_per_year):
 
-    Parameters:
-        event_impacts : impact object
-            impact object containing impacts per event
-        sampling_dict : dict
-            The sampling dictionary containing two arrays:
-                selected_events (array) : sampled events (len: total amount of sampled events)
-                events_per_year (array) : events per sampled year
+        probab_dis = freqs/np.sum(freqs)
 
-    Returns:
-        impact_per_year: array
-            Sampled impact per year (length = n_sampled_years)
-      """
+        rng = default_rng()
+        selected_events = rng.choice(indices, size=amount_events, replace=False,
+                                     p=probab_dis).astype('int')
+
+        if check_doubling: #check if an event occurs more than once in a year
+            unique_events = np.unique(selected_events)
+            #resample until each event occurs max. once per year
+            while len(unique_events) != len(selected_events):
+                rng = default_rng()
+                selected_events = rng.choice(indices, size=amount_events, replace=False,
+                                             p=probab_dis).astype('int')
+            check_doubling = False
+
+        idx_to_remove = [] #determine used events to remove them from sampling pool
+        for event in selected_events:
+            idx_to_remove.append(np.where(indices == event)[0][0])
```

Comment:
Almost always when there is a for ... append you can replace it with a list comprehension.

```suggestion
        idx_to_remove = [np.where(indices == event)[0][0] for event in selected_events]
        #determine used events to remove them from sampling pool
```